### PR TITLE
Reserve more keywords for future proofing

### PR DESCRIPTION
**Pull request by turbolent**
## Description

Code written in Cadence 1.0 must stay backward-compatible. At the same time we also want to keep evolving the language.

Conservatively reserve some more keywords we might use in the future. Adding these keywords does not imply we will use them, it just ensures we can, because already code written for 1.0 didn't use them.

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/parser/keyword.go [hunk 1]
```diff
@@ -75,6 +75,22 @@ const (
...
+	KeywordGoto        = "goto"
+	KeywordConst       = "const"
+	KeywordExport      = "export"
+	KeywordThrow       = "throw"
+	KeywordThrows      = "throws"
+	KeywordRequires    = "requires"
+	KeywordWhere       = "where"
+	KeywordFinal       = "final"
+	KeywordInternal    = "internal"
+	KeywordTypealias   = "typealias"
```
Comment (dsainati1) [inline runtime/parser/keyword.go:90]: This one seems a bit odd to me, why reserve `typealias` but not `type`?

Comment (turbolent) [reply runtime/parser/keyword.go:90]: We can also reserve `type`. I just tried to come up with a list, looking at reserved keywords of Swift, JS, Java, etc. and hadn't thought of `type`. It's not intentionally omitted

Comment (dsainati1) [reply runtime/parser/keyword.go:90]: I think `type` is commonly used to create a type alias (e.g. Go, TypeScript), so we should reserve it just to be safe

**Review by SupunS — APPROVED** [review]

File: runtime/parser/keyword.go (hunk 1 above)
Comment (turbolent) [reply runtime/parser/keyword.go:90]: I tried adding `type`, but it is used in the standard library, e.g. in `forEachStored` as an argument label. I've added it to the soft keywords set.

**Review by dsainati1 — APPROVED** [review]

**Pull request merged**